src/hxDaedalus/graphics/Pixels.hx

```diff
@@ -71,48 +83,88 @@ abstract Pixels(PixelsData)
 		var g = (value >> 8) & 0xFF;
 		var b = (value) & 0xFF;
 
-		this.bytes.set((pos + 0), a);
-		this.bytes.set((pos + 1), r);
-		this.bytes.set((pos + 2), g);
-		this.bytes.set((pos + 3), b);
+		this.bytes.set((pos + this.format.A), a);
+		this.bytes.set((pos + this.format.R), r);
+		this.bytes.set((pos + this.format.G), g);
+		this.bytes.set((pos + this.format.B), b);
+	}
+	
+	public function clone():Pixels {
+		var clone:Pixels = new Pixels(this.width, this.height, true);
+		clone.bytes.blit(0, this.bytes, 0, this.bytes.length);
+		return clone;
 	}
-
 	
-#if (flambe) // in flambe texture bytes are in RGBA format, and we want ARGB
+#if (flambe) // in flambe texture bytes are in RGBA format
 
-	@:from static public function fromTexture(texture:flambe.display.Texture) {
+	@:from static public function fromFlambeTexture(texture:flambe.display.Texture) {
 		var pixels = new Pixels(texture.width, texture.height, false);
+		pixels.format = PixelFormat.RGBA;
 		
-		// read pixels bytes in RGBA and then convert them in place to ARGB
 		pixels.bytes = texture.readPixels(0, 0, texture.width, texture.height);
-		Converter.RGBA2ARGB(pixels.bytes);
 		
 		return pixels;
 	}
 	
 	#if (flambe && html) // not possible in (flambe && flash) due to Stage3D limitations
-	public function applyTo(texture:flambe.display.Texture) {
-		var bytesRGBA = Bytes.alloc(this.width * this.height);
-		Converter.ARGB2RGBA(this.bytes, bytesRGBA);
-		texture.writePixels(bytesRGBA, 0, 0, this.width, this.height);
+	public function applyToFlambeTexture(texture:flambe.display.Texture) {
+		texture.writePixels(this.bytes, 0, 0, this.width, this.height);
 	}
 	#end
+
+#end
+
+#if (snow || luxe) // in snow/luxe texture bytes are in RGBA format (and must account for power_of_two sizes)
 	
-#elseif (flash || openfl || nme || (flambe && flash))
+	@:from static public function fromLuxeTexture(texture:phoenix.Texture) {
+		var pixels = new Pixels(texture.width, texture.height, true);
+		pixels.format = PixelFormat.RGBA;
+		
+		var pot_w = texture.width_actual;
+		var data:snow.io.typedarray.Uint8Array = texture.asset.image.data;
+		
+		var i = 0;
+		for (y in 0...pixels.height) {
+			for (x in 0...pixels.width) {
+				var pos:Int = (x << 2) + (y << 2) * pot_w;
+				pixels.bytes.set(i++, data[pos + 0]);
```

Comment:
Oh thanks, didn't know about `data.toBytes()`, that could be handy. 

Anyway I was doing it in a similar fashion in a [previous version](https://github.com/hxDaedalus/hxDaedalus/blob/a29aea06b4981b591636b1da9e9a644703294d66/src/hxDaedalus/graphics/Pixels.hx#L117-L140) (i.e. just copying the bytes over). The problem with that was that, due to POT sizes, the Potrace algorithm then examined also the bytes not in the actual texture, resulting in an unwanted contour of the original image. A solution would have been to zero out the external pixels after the copy, but I decided to go with copying just the relevant bytes, as it seemed faster.